### PR TITLE
pkg/driver_cryptocell_310: Fix Makefile

### PR DESCRIPTION
**Pull request by mguetschow**
### Contribution description

- make sure to download/extract during prepare, instead of build
- this fixes the issue of missing include dependencies for other pkgs at build time

### Testing procedure

- for an app Makefile, include both `c25519` and `driver_cryptocell_310`
- on `master`, `make all` fails with a missing include path from the `driver_cryptocell_310` package
- with this change, it works

### Issues/PRs references

- isolated from #19954 as suggested by @miri64

### CODE REVIEW

**Review by miri64 — COMMENTED** [review]
One minor nit, otherwise ready to merge IMHO

File: pkg/c25519/Makefile [hunk 1]
```diff
@@ -7,6 +7,7 @@ PKG_SHA512  = dbfb4285837ab2ea3d99c448b22877cc7a139ccbaebb1de367e2bec1fd562fe629
```
[the remark this answers is not in the document]
Comment (mguetschow) [reply pkg/c25519/Makefile:7]: Done :)

**Review by miri64 — APPROVED** [review]
ACK.

On current master

```console
$ BOARD=nrf52840dk USEPKG="c25519 driver_cryptocell_310" make -C examples/hello-world/ -j
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world'
Building application "hello-world" for "nrf52840dk" with MCU "nrf52".

make[1]: Nothing to be done for 'prepare'.
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/pkg/c25519/ 
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/c25519/src -f /home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base MODULE=c25519
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/morph25519.o] Error 1
make[2]: *** Waiting for unfinished jobs....
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/c25519.o] Error 1
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/edsign.o] Error 1
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/ed25519.o] Error 1
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/sha512.o] Error 1
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/fprime.o] Error 1
cc1: error: /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/driver_cryptocell_310/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make[2]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:146: /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/c25519/f25519.o] Error 1
make[1]: *** [Makefile:20: all] Error 2
make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/../../Makefile.include:827: pkg-build] Error 2
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world'
```

With this PR:

```
BOARD=nrf52840dk USEPKG="c25519 driver_cryptocell_310" make -C examples/hello-world/ -j
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world'
Building application "hello-world" for "nrf52840dk" with MCU "nrf52".

make[1]: Nothing to be done for 'prepare'.
mkdir -p /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg
failed: No route to host.
failed: No route to host.
2023-09-29 15:43:41 URL:https://nsscprodmedia.blob.core.windows.net/prod/software-and-other-downloads/sdks/nrf5/binaries/nrf5_sdk_17.1.0_ddde560.zip [131838843/131838843] -> "/home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/nRF5_SDK_17.1.0_ddde560.zip" [1]
test "cb610d724d27ef79c792511d8b889cbf2d411a3a391f5c740ea10b86139201b97eedf0c1edcf4d199c3fbd51e4cc5a4bd922c0c40d01e2f5359acd16a2b163e4  /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/nRF5_SDK_17.1.0_ddde560.zip" =  "$(sha512sum "/home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/nRF5_SDK_17.1.0_ddde560.zip")"
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/pkg/c25519/ 
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/build/pkg/c25519/src -f /home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base MODULE=c25519
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/pkg/cmsis/ 
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/pkg/driver_cryptocell_310/ 
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/boards/common/init
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/boards/nrf52840dk
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/core
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/core/lib
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/nrf52
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/drivers
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/pkg/driver_cryptocell_310/contrib
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/boards/common/nrf52xxxdk
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/auto_init
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/div
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/cortexm_common
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/libc
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/drivers/periph_common
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/nrf52/periph
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/malloc_thread_safe
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/newlib_syscalls_default
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/nrf52/vectors
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/nrf5x_common
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/preprocessor
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/sys/stdio_uart
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/nrf5x_common/periph
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/cpu/cortexm_common/periph
   text	   data	    bss	    dec	    hex	filename
  15488	    100	   9032	  24620	   602c	/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/nrf52840dk/hello-world.elf
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world'
```

**Review by Einhornhool — APPROVED** [review]
Looks fine, works for me

**Comment by benpicco** [comment]
bors cancel
bors merge

**Pull request merged**